indico/util/user.py

```diff
@@ -170,5 +191,16 @@ def principal_from_identifier(identifier, allow_groups=False, allow_external_use
         if event_role is None or event_role.event_id != event_id:
             raise ValueError('Invalid event role: {}'.format(event_role_id))
         return event_role
+    elif type_ == 'CategoryRole':
+        if not allow_category_roles:
+            raise ValueError('Category roles are not allowed')
+        try:
+            category_role_id = int(data)
+        except ValueError:
+            raise ValueError('Invalid data')
+        category_role = CategoryRole.get(category_role_id)
+        if category_role is None:
+            raise ValueError('Invalid category role: {}'.format(category_role_id))
```

Comment:
We also need to check whether the role is available here (based on the category of the event from `event_id`).